Add validation to autoyast_mini scenario

jknphy opened the pull request:
- Related ticket: https://progress.opensuse.org/issues/68773
- Job Group MR: https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/244
- Verification run: https://openqa.suse.de/tests/overview?build=jknphy%2Fos-autoinst-distri-opensuse%23validate_ay_mini&distri=sle&version=15-SP2

b10n1k commented on tests/console/validate_file_system.pm:
```diff
@@ -54,6 +55,11 @@ sub run {
               sub { m/Version:\s+$luks_type.*/s };
         }
     }
+    if (defined $swap) {
```
i have the impression this will be always true with `defined`, whatever the value of the $swap. i assume that you expect that this will defined in the yaml or not. Assuming that, this will work as you expected without problem. But maybe it would be more accurate to have something like `defined $swap && $swap`. Do i miss something?

jknphy replied:
I was repeating the pattern above, but you're right, might confuse someone, then it is more simple just `if($swap)`.

jknphy commented on tests/console/validate_file_system.pm:
```diff
@@ -54,6 +55,11 @@ sub run {
     }
+    if ($swap) {
+        record_info("Check swap", "Check swap partition exists");
+        assert_script_run("lsblk -n | grep [SWAP]",
```
Yes I have this idea that every test module is in fact using different test_data (some times are subtle conversions) so I wanted to match the square brackets as well.
I didn't notice that bash thingy, good catch, thanks!, fixed.

jknphy commented:
> ~Ok, I see that validation of the generated profile is done, but please, don't forget about the installation itself. With autoyast we always have those 2 parts, one is the installation itself and second one is cloning of the system.~
> Sorry, got distracted. Do you think that validating filesystem is enough? Ticket was not refined, so that's my question. I believe that partitioning is not the only part we could validate with existing modules.

I also tried module for btrfs but it didn't work, I could only find this one useful for partitioning (with some adaptation). In `test autoyast/` and in `test/console/(validate|verify)_.* I could not find anything useful to reuse outside of partitioning that wouldn't be very specific, from the input profile, users and groups but I don't think we have a module for that, and if we validate the cloned one, that could be tricky because we don't have requirements and those default could be written differently in the future. Do you have something in mind that I can schedule?

rwx788 commented:
> > ~Ok, I see that validation of the generated profile is done, but please, don't forget about the installation itself. With autoyast we always have those 2 parts, one is the installation itself and second one is cloning of the system.~
> > Sorry, got distracted. Do you think that validating filesystem is enough? Ticket was not refined, so that's my question. I believe that partitioning is not the only part we could validate with existing modules.
> 
> I also tried module for [btrfs](https://openqa.suse.de/tests/4431723#step/verify_btrfs/3) but it didn't work, I could only find this one useful for partitioning (with some adaptation). In `test autoyast/` and in `test/console/(validate|verify)_.* I could not find anything useful to reuse outside of partitioning that wouldn't be very specific. From the input profile, users and groups but I don't think we have a module for that, and if we validate the cloned one, that could be tricky because we don't have requirements and those default could be written differently in the future. Do you have something in mind that I can schedule?

There are modules we could reuse from validation of the default installation, running some zypper commands at least, checking that system role selection corresponds to what we see in the system, etc. Please, don't rush to provide cheapest to deliver solution and keep in mind that we have other scenarios to reuse this functionality.

jknphy commented:
> > > ~Ok, I see that validation of the generated profile is done, but please, don't forget about the installation itself. With autoyast we always have those 2 parts, one is the installation itself and second one is cloning of the system.~
> > > Sorry, got distracted. Do you think that validating filesystem is enough? Ticket was not refined, so that's my question. I believe that partitioning is not the only part we could validate with existing modules.
> > 
> > 
> > I also tried module for [btrfs](https://openqa.suse.de/tests/4431723#step/verify_btrfs/3) but it didn't work, I could only find this one useful for partitioning (with some adaptation). In `test autoyast/` and in `test/console/(validate|verify)_.* I could not find anything useful to reuse outside of partitioning that wouldn't be very specific. From the input profile, users and groups but I don't think we have a module for that, and if we validate the cloned one, that could be tricky because we don't have requirements and those default could be written differently in the future. Do you have something in mind that I can schedule?
> 
> There are modules we could reuse from validation of the default installation, running some zypper commands at least, checking that system role selection corresponds to what we see in the system, etc. Please, don't rush to provide cheapest to deliver solution and keep in mind that we have other scenarios to reuse this functionality.

I added some modules from default installation and other test suites (currently running), some of them they are not strictly validation (validating expected vs actual with some sort of test data), which was my initial query, but those have some purpose at least to check that the tooling works. Many of them I haven't see them in years :) so no rush from my side or trying to delivering the cheapest (which is IMO this kind of delivery it is just a point of view depending on someone experience), in fact I fixed one module which is not in the scope to make it work.
 
I was just wrongly assuming that the ticket didn't require further refinement and in general in the way that the modules have been designed for years we could not reuse much without further changes.
Anyway, if you see more interesting modules let me know and I can test and run them to see if they fit.

note: I was not able to find anything about system role validation.

jknphy commented:
> So I would propose to grab something from this test suite: https://openqa.suse.de/tests/4445795
> like `ncurses`, `firewall_enabled`, `orphaned_packages_check`, maybe `sshd` depending if we have something more lightweight.

firewall tools are not there: https://openqa.suse.de/tests/4447062#step/firewall_enabled/1
sshd fails here: https://openqa.suse.de/tests/4449871#step/sshd/30
The other suggestion are added (running verification)

rwx788 commented:
> > So I would propose to grab something from this test suite: https://openqa.suse.de/tests/4445795
> > like `ncurses`, `firewall_enabled`, `orphaned_packages_check`, maybe `sshd` depending if we have something more lightweight.
> 
> firewall tools are not there: https://openqa.suse.de/tests/4447062#step/firewall_enabled/1
> sshd fails here: https://openqa.suse.de/tests/4449871#step/sshd/30
> The other suggestion are added (running verification)

I guess we should implement some test module for sanity checks for the default installation, checking most critical services states, but let's leave it for later.

Could you please check what didn't work on powerVM?

jknphy commented:
> > > So I would propose to grab something from this test suite: https://openqa.suse.de/tests/4445795
> > > like `ncurses`, `firewall_enabled`, `orphaned_packages_check`, maybe `sshd` depending if we have something more lightweight.
> > 
> > 
> > firewall tools are not there: https://openqa.suse.de/tests/4447062#step/firewall_enabled/1
> > sshd fails here: https://openqa.suse.de/tests/4449871#step/sshd/30
> > The other suggestion are added (running verification)
> 
> I guess we should implement some test module for sanity checks for the default installation, checking most critical services states, but let's leave it for later.
> 
> Could you please check what didn't work on powerVM?

Yes, sound good for the future.
Maybe it failed because I was using alternative SCC_URL, now is fine.
